go/vt/sqlparser/analyzer.go

```diff
@@ -31,13 +31,16 @@ func IsColName(node ValExpr) bool {
 // IsValue returns true if the ValExpr is a string, integral or value arg.
 // NULL is not considered to be a value.
 func IsValue(node ValExpr) bool {
-	v, ok := node.(*SQLVal)
-	if !ok {
-		return false
-	}
-	switch v.Type {
-	case StrVal, HexVal, IntVal, ValArg:
-		return true
+	switch v := node.(type) {
```

Comment:
Since we're now resolving the ValuesFuncExpr into a real value, we need to try operating on its resolved value, if it exists. Both here and in AsInterface below